<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />

<meta name="author" content="Audrey Hendricks, audrey.hendricks@ucdenver.edu" />
<meta name="author" content="Gregory Matesi, gregory.matesi@ucdenver.edu" />

<meta name="date" content="2020-11-01" />

<title>Summix: estimating and adjusting for ancestry in genetic summary data</title>

<style type="text/css">
a.anchor-section {margin-left: 10px; visibility: hidden; color: inherit;}
a.anchor-section::before {content: '#';}
.hasAnchor:hover a.anchor-section {visibility: visible;}
</style>
<script>// Anchor sections v1.0 written by Atsushi Yasumoto on Oct 3rd, 2020.
document.addEventListener('DOMContentLoaded', function() {
  // Do nothing if AnchorJS is used
  if (typeof window.anchors === 'object' && anchors.hasOwnProperty('hasAnchorJSLink')) {
    return;
  }

  const h = document.querySelectorAll('h1, h2, h3, h4, h5, h6');

  // Do nothing if sections are already anchored
  if (Array.from(h).some(x => x.classList.contains('hasAnchor'))) {
    return null;
  }

  // Use section id when pandoc runs with --section-divs
  const section_id = function(x) {
    return ((x.classList.contains('section') || (x.tagName === 'SECTION'))
            ? x.id : '');
  };

  // Add anchors
  h.forEach(function(x) {
    const id = x.id || section_id(x.parentElement);
    if (id === '') {
      return null;
    }
    let anchor = document.createElement('a');
    anchor.href = '#' + id;
    anchor.classList = ['anchor-section'];
    x.classList.add('hasAnchor');
    x.appendChild(anchor);
  });
});
</script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(title);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>



<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">Summix: estimating and adjusting for ancestry in genetic summary data</h1>
<h4 class="author">Audrey Hendricks, <a href="mailto:audrey.hendricks@ucdenver.edu" class="email">audrey.hendricks@ucdenver.edu</a></h4>
<h4 class="author">Gregory Matesi, <a href="mailto:gregory.matesi@ucdenver.edu" class="email">gregory.matesi@ucdenver.edu</a></h4>
<h4 class="date">November 1, 2020</h4>


<div id="TOC">
<ul>
<li><a href="#introduction"><span class="toc-section-number">1</span> Introduction</a></li>
<li><a href="#summix"><span class="toc-section-number">2</span> Summix</a><ul>
<li><a href="#slsqp"><span class="toc-section-number">2.1</span> SLSQP</a></li>
<li><a href="#usage"><span class="toc-section-number">2.2</span> Usage</a><ul>
<li><a href="#arguments"><span class="toc-section-number">2.2.1</span> Arguments</a></li>
<li><a href="#details"><span class="toc-section-number">2.2.2</span> Details</a></li>
<li><a href="#value"><span class="toc-section-number">2.2.3</span> Value</a></li>
</ul></li>
<li><a href="#example"><span class="toc-section-number">2.3</span> Example</a></li>
<li><a href="#example-1"><span class="toc-section-number">2.4</span> Example</a></li>
</ul></li>
<li><a href="#adjaf"><span class="toc-section-number">3</span> adjAF</a><ul>
<li><a href="#usage-1"><span class="toc-section-number">3.1</span> Usage</a><ul>
<li><a href="#arguments-1"><span class="toc-section-number">3.1.1</span> Arguments</a></li>
<li><a href="#details-1"><span class="toc-section-number">3.1.2</span> Details</a></li>
<li><a href="#value-1"><span class="toc-section-number">3.1.3</span> Value</a></li>
</ul></li>
<li><a href="#example-2"><span class="toc-section-number">3.2</span> Example</a></li>
</ul></li>
</ul>
</div>

<style type="text/css">

h1.title {
  font-size: 38px;
  color: Black;
  text-align: center;
}
h4.author { /* Header 4 - and the author and data headers use this too  */
    font-size: 18px;
  ##font-family: "Times New Roman", Times, serif;
  color: Black;
  text-align: center;
}

h4.date { /* Header 4 - and the author and data headers use this too  */
  font-size: 18px;
  ##font-family: "Times New Roman", Times, serif;
  color: Black;
  text-align: center;
}

div#TOC li {
    list-style:none;
    background-image:none;
    background-repeat:none;
    background-position:0; 
}
</style>
<div id="introduction" class="section level1">
<h1><span class="header-section-number">1</span> Introduction</h1>
<p>Hidden heterogeneity (such as ancestry) within genetic summary data can lead to confounding in association testing or inaccurate prioritization of putative variants. Here, we provide Summix, a method to estimate and adjust for reference ancestry groups within genetic allele frequency data. This method was developed by the Summix team at the University of Colorado Denver and is headed by Dr Audrey Hendricks.</p>
<p><em>References</em> Cite the Summix paper</p>
</div>
<div id="summix" class="section level1">
<h1><span class="header-section-number">2</span> Summix</h1>
<p>Function to estimate reference ancestry proportions in heterogeneous genetic data.</p>
<div id="slsqp" class="section level2">
<h2><span class="header-section-number">2.1</span> SLSQP</h2>
<p>The Summix function uses the slsqp() function in the nloptr package to run Sequential Quadratic Programming. <a href="https://www.rdocumentation.org/packages/nloptr/versions/1.2.2.2/topics/slsqp" class="uri">https://www.rdocumentation.org/packages/nloptr/versions/1.2.2.2/topics/slsqp</a></p>
</div>
<div id="usage" class="section level2">
<h2><span class="header-section-number">2.2</span> Usage</h2>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" title="1"><span class="kw">summix</span>(data, reference, observed, pi.start)</a></code></pre></div>
<div id="arguments" class="section level3">
<h3><span class="header-section-number">2.2.1</span> Arguments</h3>
<p><strong>data</strong> Data frame of the observed and reference allele frequencies for N genetic variants. See data formatting document at <a href="https://github.com/hendriau/Summix" class="uri">https://github.com/hendriau/Summix</a> for more information</p>
<p><strong>reference</strong> Character vector of the column names for the reference ancestries.</p>
<p><strong>observed</strong> Column name of the heterogeneous observed ancestry as a character string,</p>
<p><strong>pi.start</strong> Numeric vector of length K of the starting guess for the ancestry proportions. If not specified, this defaults to 1/K where K is the number of reference ancestry groups.</p>
</div>
<div id="details" class="section level3">
<h3><span class="header-section-number">2.2.2</span> Details</h3>
<p>Estimates the proportion of each reference ancestry within the chosen observed group</p>
</div>
<div id="value" class="section level3">
<h3><span class="header-section-number">2.2.3</span> Value</h3>
<p>Data frame with components:</p>
<div class="line-block"><strong>objective</strong> Least Squares value at solution<br />
<br />
<strong>iterations</strong> Number of iterations<br />
<br />
<strong>time</strong> Function run time in seconds<br />
<br />
<strong>filtered</strong> Number of NA SNPs filtered out<br />
<br />
<strong>K estimated proportions</strong> Total of K estimated ancestry proportions.</div>
</div>
</div>
<div id="example" class="section level2">
<h2><span class="header-section-number">2.3</span> Example</h2>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" title="1"><span class="kw">library</span>(<span class="st">&quot;Summix&quot;</span>)</a>
<a class="sourceLine" id="cb2-2" title="2"><span class="co"># load the data</span></a>
<a class="sourceLine" id="cb2-3" title="3"><span class="kw">data</span>(<span class="st">&quot;ancestryData&quot;</span>)</a>
<a class="sourceLine" id="cb2-4" title="4"> </a>
<a class="sourceLine" id="cb2-5" title="5"><span class="co"># Estimate 5 reference ancestry proportion values for the gnomAD African/African Amercian ancestry group</span></a>
<a class="sourceLine" id="cb2-6" title="6"><span class="kw">summix</span>( <span class="dt">data =</span> ancestryData, </a>
<a class="sourceLine" id="cb2-7" title="7">      <span class="dt">reference=</span><span class="kw">c</span>(<span class="st">&quot;ref_AF_afr_1000G&quot;</span>, </a>
<a class="sourceLine" id="cb2-8" title="8">          <span class="st">&quot;ref_AF_eur_1000G&quot;</span>, </a>
<a class="sourceLine" id="cb2-9" title="9">          <span class="st">&quot;ref_AF_sas_1000G&quot;</span>, </a>
<a class="sourceLine" id="cb2-10" title="10">          <span class="st">&quot;ref_AF_iam_1000G&quot;</span>, </a>
<a class="sourceLine" id="cb2-11" title="11">          <span class="st">&quot;ref_AF_eas_1000G&quot;</span>), </a>
<a class="sourceLine" id="cb2-12" title="12">      <span class="dt">observed=</span><span class="st">&quot;gnomad_AF_afr&quot;</span> )</a></code></pre></div>
<pre><code>##   objective iterations           time filtered ref_AF_afr_1000G
## 1  2.135725         35 0.6270061 secs        0        0.8250554
##   ref_AF_eur_1000G ref_AF_sas_1000G ref_AF_iam_1000G ref_AF_eas_1000G
## 1        0.1576768      0.003285205      0.006308355      0.007674304</code></pre>
</div>
<div id="example-1" class="section level2">
<h2><span class="header-section-number">2.4</span> Example</h2>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" title="1"><span class="kw">library</span>(<span class="st">&quot;Summix&quot;</span>)</a>
<a class="sourceLine" id="cb4-2" title="2"><span class="co"># load the data</span></a>
<a class="sourceLine" id="cb4-3" title="3"><span class="kw">data</span>(<span class="st">&quot;ancestryData&quot;</span>)</a>
<a class="sourceLine" id="cb4-4" title="4"> </a>
<a class="sourceLine" id="cb4-5" title="5"><span class="co"># Estimate 5 reference ancestry proportion values for the gnomAD African/African Amercian ancestry group</span></a>
<a class="sourceLine" id="cb4-6" title="6"><span class="kw">summix</span>( <span class="dt">data =</span> ancestryData, </a>
<a class="sourceLine" id="cb4-7" title="7">      <span class="dt">reference=</span><span class="kw">c</span>(<span class="st">&quot;ref_AF_afr_1000G&quot;</span>, </a>
<a class="sourceLine" id="cb4-8" title="8">          <span class="st">&quot;ref_AF_eur_1000G&quot;</span>, </a>
<a class="sourceLine" id="cb4-9" title="9">          <span class="st">&quot;ref_AF_sas_1000G&quot;</span>, </a>
<a class="sourceLine" id="cb4-10" title="10">          <span class="st">&quot;ref_AF_iam_1000G&quot;</span>, </a>
<a class="sourceLine" id="cb4-11" title="11">          <span class="st">&quot;ref_AF_eas_1000G&quot;</span>), </a>
<a class="sourceLine" id="cb4-12" title="12">      <span class="dt">observed=</span><span class="st">&quot;gnomad_AF_afr&quot;</span>,</a>
<a class="sourceLine" id="cb4-13" title="13">      <span class="dt">pi.start =</span> <span class="kw">c</span>(<span class="fl">0.8</span>, <span class="fl">0.1</span>, <span class="fl">0.05</span>, <span class="fl">0.02</span>, <span class="fl">0.03</span>))</a></code></pre></div>
<pre><code>##   objective iterations           time filtered ref_AF_afr_1000G
## 1  2.135725         29 0.5800009 secs        0        0.8250554
##   ref_AF_eur_1000G ref_AF_sas_1000G ref_AF_iam_1000G ref_AF_eas_1000G
## 1        0.1576768      0.003285205      0.006308355      0.007674304</code></pre>
</div>
</div>
<div id="adjaf" class="section level1">
<h1><span class="header-section-number">3</span> adjAF</h1>
<p><em>Ancestry Adjusted Allele Frequency</em> Function to estimate ancestry adjusted allele frequencies given the proportion of reference ancestry groups.</p>
<div id="usage-1" class="section level2">
<h2><span class="header-section-number">3.1</span> Usage</h2>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" title="1"><span class="kw">adjAF</span>(data, reference, observed, pi.target, pi.observed)</a></code></pre></div>
<div id="arguments-1" class="section level3">
<h3><span class="header-section-number">3.1.1</span> Arguments</h3>
<p><strong>data</strong> Data frame of unadjusted allele frequency for observed group, K-1 reference ancestry allele frequencies for N SNPs</p>
<p><strong>reference</strong> Character vector of the column names for K-1 reference ancestry groups. The name of the last reference ancestry group is not included as that group is not used to estimate the adjusted allele frequencies.</p>
<p><strong>observed</strong> Column name for the observed ancestry.</p>
<p><strong>pi.observed</strong> Numeric vector of the mixture proportions for K reference ancestry groups for the observed group. The order must match the order of the reference specified reference character vector with the last entry matching the missing ancestry reference group.</p>
<p><strong>pi.target</strong> Numeric vector of the mixture proportions for K reference ancestry groups in the target sample or subject. Order must match the order of the specified reference character vector with the last entry matching the missing ancestry reference group.</p>
</div>
<div id="details-1" class="section level3">
<h3><span class="header-section-number">3.1.2</span> Details</h3>
<p>Estimates ancestry adjusted allele frequencies in an observed sample of allele frequencies given estimated reference ancestry proportions and the observed AFs for K-1 reference ancestry groups.</p>
</div>
<div id="value-1" class="section level3">
<h3><span class="header-section-number">3.1.3</span> Value</h3>
<p>List with components:</p>
<div class="line-block"><strong>pi</strong> table of input reference ancestry groups, pi.observed values, and pi.target values<br />
<br />
<strong>observed.data</strong> name of the data column for the observed group from which adjusted ancestry allele frequency is estimated<br />
<br />
<strong>Nsnps</strong> number of SNPs for which adjusted AF is estimated<br />
<br />
<strong>adjusted.AF</strong> data frame of original data with an appended column of adjusted allele frequencies</div>
</div>
</div>
<div id="example-2" class="section level2">
<h2><span class="header-section-number">3.2</span> Example</h2>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" title="1"><span class="kw">library</span>(<span class="st">&quot;Summix&quot;</span>)</a>
<a class="sourceLine" id="cb7-2" title="2"><span class="kw">data</span>(ancestryData)</a>
<a class="sourceLine" id="cb7-3" title="3"></a>
<a class="sourceLine" id="cb7-4" title="4"><span class="kw">head</span>(ancestryData)</a></code></pre></div>
<pre><code>##   CHR       RSID     POS A1 A2 ref_AF_eur_1000G ref_AF_afr_1000G
## 1   1  rs2887286 1156131  C  T      0.173275495       0.54166349
## 2   1 rs41477744 2329564  A  G      0.001237745       0.03571448
## 3   1  rs9661525 2952840  G  T      0.168316089       0.12004821
## 4   1  rs2817174 3044181  C  T      0.428212624       0.95932526
## 5   1 rs12139206 3504073  T  C      0.204214851       0.80156548
## 6   1  rs7514979 3654595  T  C      0.004950604       0.41865218
##   ref_AF_sas_1000G ref_AF_eas_1000G ref_AF_iam_1000G gnomad_AF_afr
## 1       0.53171227        0.8462232           0.7093     0.4886100
## 2       0.00000000        0.0000000           0.0000     0.0459137
## 3       0.09918029        0.3938534           0.2442     0.1359770
## 4       0.63907198        0.5704540           0.5000     0.8548790
## 5       0.39367076        0.3898812           0.3372     0.7241780
## 6       0.00000000        0.0000000           0.0000     0.3362490
##   gnomad_AF_amr gnomad_AF_oth
## 1    0.52594300    0.22970500
## 2    0.00117925    0.00827206
## 3    0.28605200    0.15561700
## 4    0.48818000    0.47042500
## 5    0.29550800    0.25874800
## 6    0.01650940    0.02481620</code></pre>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" title="1">tmp.aa&lt;-<span class="kw">adjAF</span>(<span class="dt">data   =</span> ancestryData,</a>
<a class="sourceLine" id="cb9-2" title="2">     <span class="dt">reference   =</span> <span class="kw">c</span>(<span class="st">&quot;ref_AF_eur_1000G&quot;</span>),</a>
<a class="sourceLine" id="cb9-3" title="3">     <span class="dt">observed    =</span> <span class="st">&quot;gnomad_AF_afr&quot;</span>,</a>
<a class="sourceLine" id="cb9-4" title="4">     <span class="dt">pi.target   =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">1</span>),</a>
<a class="sourceLine" id="cb9-5" title="5">     <span class="dt">pi.observed =</span> <span class="kw">c</span>(.<span class="dv">15</span>, <span class="fl">.85</span>))</a></code></pre></div>
<pre><code>## $pi
##          ref.group pi.observed pi.target
## 1 ref_AF_eur_1000G        0.15         0
## 2             NONE        0.85         1
## 
## $observed.data
## [1] &quot;observed data to update AF: &#39;gnomad_AF_afr&#39;&quot;
## 
## $Nsnps
## [1] 10000
## 
## [[4]]
## [1] &quot;use $adjusted.AF to see adjusted AF data&quot;</code></pre>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" title="1">tmp.aa<span class="op">$</span>adjusted.AF[<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>,]</a></code></pre></div>
<pre><code>##   CHR       RSID     POS A1 A2 ref_AF_eur_1000G ref_AF_afr_1000G
## 1   1  rs2887286 1156131  C  T      0.173275495       0.54166349
## 2   1 rs41477744 2329564  A  G      0.001237745       0.03571448
## 3   1  rs9661525 2952840  G  T      0.168316089       0.12004821
## 4   1  rs2817174 3044181  C  T      0.428212624       0.95932526
## 5   1 rs12139206 3504073  T  C      0.204214851       0.80156548
##   ref_AF_sas_1000G ref_AF_eas_1000G ref_AF_iam_1000G gnomad_AF_afr
## 1       0.53171227        0.8462232           0.7093     0.4886100
## 2       0.00000000        0.0000000           0.0000     0.0459137
## 3       0.09918029        0.3938534           0.2442     0.1359770
## 4       0.63907198        0.5704540           0.5000     0.8548790
## 5       0.39367076        0.3898812           0.3372     0.7241780
##   gnomad_AF_amr gnomad_AF_oth adjustedAF
## 1    0.52594300    0.22970500 0.54425727
## 2    0.00117925    0.00827206 0.05379769
## 3    0.28605200    0.15561700 0.13027010
## 4    0.48818000    0.47042500 0.93017307
## 5    0.29550800    0.25874800 0.81593620</code></pre>
</div>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
